### ISSUE TITLE
Upgrade to 4.5-RC1 failed due to "Multiple input file properties with name '$1'"

### ISSUE THREAD

**Issue by tshasc** [gradle/gradle#4085]
I tried to upgrade Gradle from 4.3 to 4.5-RC1, I got below error when I run below command.
seems this error related to https://github.com/gradle/gradle/pull/3720, how can I fix the issue and upgrade to 4.5-RC1 successfully?

@wolfs 

**Command**
./gradlew -S :proto-converter:generateProto

**Result**

```
The :proto-converter:generateProto task failed.
Multiple input file properties with name '$1'
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':proto-converter:generateProto'.Open stacktrace
Caused by: java.lang.IllegalArgumentException: Multiple input file properties with name '$1'Close stacktrace
at org.gradle.api.internal.tasks.properties.GetInputFilesVisitor.visitInputFileProperty(GetInputFilesVisitor.java:44)
at org.gradle.api.internal.tasks.properties.CompositePropertyVisitor.visitInputFileProperty(CompositePropertyVisitor.java:35)
at org.gradle.api.internal.tasks.DefaultTaskInputs.visitRegisteredProperties(DefaultTaskInputs.java:76)
at org.gradle.api.internal.tasks.TaskPropertyUtils.visitProperties(TaskPropertyUtils.java:39)
at org.gradle.api.internal.tasks.execution.DefaultTaskProperties.resolve(DefaultTaskProperties.java:77)
at org.gradle.execution.taskgraph.TaskInfo.getTaskProperties(TaskInfo.java:229)
at org.gradle.execution.taskgraph.TaskInfo.getDestroyables(TaskInfo.java:220)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.getDestroyablePaths(DefaultTaskExecutionPlan.java:794)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.canRunWithCurrentlyExecutedTasks(DefaultTaskExecutionPlan.java:663)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.access$2600(DefaultTaskExecutionPlan.java:89)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan$9.transform(DefaultTaskExecutionPlan.java:593)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan$9.transform(DefaultTaskExecutionPlan.java:588)
at org.gradle.internal.resources.DefaultResourceLockCoordinationService.withStateLock(DefaultResourceLockCoordinationService.java:45)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.selectNextTask(DefaultTaskExecutionPlan.java:588)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.access$2300(DefaultTaskExecutionPlan.java:89)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan$8.transform(DefaultTaskExecutionPlan.java:562)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan$8.transform(DefaultTaskExecutionPlan.java:543)
at org.gradle.internal.resources.DefaultResourceLockCoordinationService.withStateLock(DefaultResourceLockCoordinationService.java:45)
at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:543)
at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
at java.lang.Thread.run(Thread.java:748)
```

**Comment by bmuschko** [comment]
Can you please provide an example project?

**Comment by tshasc** [comment]
@bmuschko 
1. I will try to provide an example project.
2. The exception came from below, I wonder could u give me an example for possible inputFileProperty/propertyName so that I can check whether my code may cause the issue

 public void visitInputFileProperty(TaskInputFilePropertySpec inputFileProperty) {
        String propertyName = inputFileProperty.getPropertyName();
        if (!names.add(propertyName)) {
            throw new IllegalArgumentException(String.format("Multiple %s file properties with name '%s'", "input", propertyName));
        }
3. I have no idea what is $1, let me try to understand what u mean for your last sentence.

**Comment by eriwen** [comment]
@tshasc It would help us narrow the problem if you would try to reproduce this on 4.4.1 to understand if the regression exists in 4.5. Additionally, a [build scan](https://scans.gradle.com) (if you can get one) would provide lots of helpful configuration and environment information.

**Comment by tshasc** [comment]
@eriwen 
in fact, I have a scan result, sorry I don't want to share everything at this moment.
but I am curious which section may help you for the investigation, I can try to provide that part

I updated to 4.4.1 and  "./gradlew -S :proto-converter:generateProto" works for me
./gradlew wrapper --gradle-version=4.4.1

Seems this is a regression?

**Comment by eriwen** [comment]
@tshasc Thanks for the additional information.

Regarding a build scan, I understand your hesitation around IP. The "Switches" and "Infrastructure properties" and most important for tracking this down because they help us narrow down the particular configuration and environment so we can reproduce this ourselves. 

My understanding is that someone from our build cache team is looking at this, so we'll hopefully get it fixed for 4.5.

**Comment by tshasc** [comment]
@eriwen below are the two sections what you want, i wish it could help

1 active switch
Parallel 	On
Build Cache 	Off
Configure on demandEnables configuration on demand for multi-project builds, reducing overhead for decoupled projects. 	Off
Continue on failure 	Off
Continuous 	Off
Daemon 	Off
Dry run 	Off
Offline 	Off
Re-run tasks 	Off
Refresh dependencies 	Off

13 infrastructure properties
Operating system	Mac OS X 10.12.6
CPU cores	8 cores
Max Gradle workers	8 workers
Java runtime	Oracle Java(TM) SE Runtime Environment 1.8.0_152-b16
Java VM	Oracle Java HotSpot(TM) 64-Bit Server VM 25.152-b16 (mixed mode)
Max JVM memory heap size	3817 MB
Locale	English (United States)
Default charset	UTF-8
Username	***
Public hostname	***
Local hostname	(N/A)
Public IP address	***
Local IP addresses	192.168.0.*

**Comment by lptr** [comment]
Hi @tshasc, thanks for the report, I'm taking a look now. Why do you think this is related to #3720?

**Comment by lptr** [comment]
What would also be very useful for me is if I could understand a bit more how your `:proto-converter:generateProto` task is configured.

**Comment by lptr** [comment]
I tried running [this example project](https://github.com/google/protobuf-gradle-plugin/tree/master/exampleProject) with Gradle 4.5-rc-1 but it works fine, so I'm guessing the issue might be triggered by something specific in your build.

**Comment by tshasc** [comment]
@lptr 
The exception related logic was added in that PR.

how can I get "how the task is configured", what should I check?

**Comment by lptr** [comment]
Are you using the latest version of the Protobuf Gradle plugin (0.8.3)?

**Comment by tshasc** [comment]
we are using 0.8.0 now, should I upgrade that?

**Comment by lptr** [comment]
I think I found the problem, trying to reproduce now.

**Comment by lptr** [comment]
Yep, reproduction works, trying to fix.

**Comment by tshasc** [comment]
cool, let me know when there is a formal build for the fix so that I can try again :)

**Comment by big-guy** [comment]
@tshasc Could you give this nightly a try?

`gradlew wrapper --gradle-version=4.5-20180120003357+0000`

**Comment by tshasc** [comment]
if works if I am using above version, thanks

**Comment by lptr** [comment]
Thanks. The fix will ship in 4.5-rc-2.
